repo.go

```diff
@@ -467,11 +467,22 @@ func (r *Repo) RevokeKeyWithExpires(keyRole, id string, expires time.Time) error
 		return ErrKeyNotFound{keyRole, id}
 	}
 	role.KeyIDs = keyIDs
+	root.Roles[keyRole] = role
 
-	for _, keyID := range key.IDs() {
-		delete(root.Keys, keyID)
+	// Only delete the key from root.Keys if no other role is using that key.
+	delete_key := true
```

Comment:
nit: please consider renaming delete_key-> key_in_use to make the logic more readable.